cmd/maya-exporter/app/command/exporter_test.go

```diff
@@ -0,0 +1,64 @@
+package command
+
+import (
+	"net"
+	"reflect"
+	"testing"
+)
+
+func TestInitialize(t *testing.T) {
+	cases := map[string]struct {
+		cmdOptions *VolumeExporterOptions
+		output     string
+	}{
+		"Storage engine is cstor": {
+			cmdOptions: &VolumeExporterOptions{
+				StorageEngine: "cstor",
+			},
+			output: "cstor",
+		},
+		"storage engine is jiva": {
+			cmdOptions: &VolumeExporterOptions{
+				StorageEngine: "jiva",
+			},
+			output: "jiva",
+		},
+		"storage engine is other": {
+			cmdOptions: &VolumeExporterOptions{
+				StorageEngine: "other",
+			},
+			output: "",
+		},
+	}
+	for name, tt := range cases {
+		t.Run(name, func(t *testing.T) {
+			got := Initialize(tt.cmdOptions)
+			if !reflect.DeepEqual(got, tt.output) {
+				t.Fatalf("Initialize() => %v, want %v", got, tt.output)
+			}
+		})
+	}
+}
+
+func TestStartMayaExporter(t *testing.T) {
+	ErrorMessage := make(chan error)
+	cmdOptions := &VolumeExporterOptions{
+		ControllerAddress: "localhost:9501",
+		MetricsPath:       "/metrics",
+		ListenAddress:     ":9500",
+	}
+	go func() {
+		//Block port 9500 and attempt to start http server at 9500.
+		p1, err := net.Listen("tcp", "localhost:9500")
+		defer p1.Close()
+		if err != nil {
+			t.Log(err)
+		}
+		ErrorMessage <- cmdOptions.StartMayaExporter()
+	}()
+	msg := <-ErrorMessage
+	if msg != nil {
+		t.Log("Try to start http server in a port which is busy.")
```

Comment:
Could not understand this message? What are we trying to achieve by logging inside a UT?

Reply:
@AmitKumarDas is there any other option ? 
As we can't write a positive test case to start an http server because that will run forever, but we can test about when it fails. Let's say the port used is already in use. So we can't call fatal because this is negative test case which should pass. 
Anyway i have changed the log msg and it is more comprehensive.
Now for testing the same (+ve test cases) i need to mock the whole method which i will do in the next iteration because this code will be changed later. So i'm not putting more efforts on mocking.